lib/kubeclient/config.rb

```diff
@@ -142,16 +149,15 @@ def fetch_user_key_data(user)
         File.read(ext_file_path(user['client-key']))
       elsif user.key?('client-key-data')
         Base64.decode64(user['client-key-data'])
+      elsif user.key?('clientKeyData')
+        user['clientKeyData']
```

Comment:
It's slightly confusing to have 2 similarly named keys, one base64, one not.
Although both spellings come from k8s.

The reuse of `user` is clever but WDYT of putting it in a sub-field instead?  `user.exec_result = ExecCredentials.run(exec_opts)`
It does mean you'll have to check .token in 2 places, but I think the data flow will be more readable.

Reply:
I like it. Done.